### PR TITLE
CEN-49/50: Two Post Request and Get Fixes

### PR DESCRIPTION
**Pull request by lamad019**
1. Add book POST request made. Multiple checks to avoid any errors.
2. Create new wishlist POST request made. Multiple checks to avoid any errors. Also accounted for a maximum of 3 wish lists per user.
3. Fixed the GET request to properly post book names for whatever wishlist ID was inputted.

### CODE REVIEW

**Comment by lamad019** [comment]
3*. Will later update the url path to take parameters rather than use the URL itself. Also will include checks to avoid any issues involving the wishlist id parameter.

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 1]
```diff
@@ -1,11 +1,146 @@
...
+        wishlist_list.append(wID.id)
+    
+    #Obtain book ID and wishlist ID from POST request
+    try:
+        data = json.loads(request.body)
+    except json.JSONDecodeError:
+        data = {}
+
+    #Obtain both IDs and set them to their associated variables for checks.
+    bookID = data.get('bookID')
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:44]:
```suggestion
    bookID = data.get('book_id')
```

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 2]
```diff
@@ -1,11 +1,146 @@
...
+    wishlistID = data.get('wishlistID')
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:45]:
```suggestion
    wishlistID = data.get('wish_list_id')
```

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 3]
```diff
@@ -1,11 +1,146 @@
...
+
+    #Check if either ID is missing in request. If so, send error. Else, continue with bookID check.
+    if bookID == None or wishlistID == None:
+        response = {
+            'Error' : 'BookID or WishlistID missing. Please enter it again.'
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:50]:
```suggestion
            'Error' : 'book_id or wish_list_id missing. Please enter it again.'
```

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 4]
```diff
@@ -1,11 +1,146 @@
...
+        }
+        return JsonResponse(response, safe=False)
+    
+    #Check for existing book ID. If not, send error. Else, continue to wishlistID check.
+    if bookID not in bookID_list:
+        response = {
+            'Error' : 'BookID submitted does not exist within the database.'
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:57]:
```suggestion
            'Error' : 'book_id submitted does not exist within the database.'
```

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 5]
```diff
@@ -1,11 +1,146 @@
...
+        }
+        return JsonResponse(response, safe=False)
+    
+    #Check for existing wishlist ID. If not, send error. Else, create new wishlist_book into associated table.
+    if wishlistID not in wishlist_list:
+        response = {
+            'Error' : 'WishlistID submitted does not exist within the database.'
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:64]:
```suggestion
            'Error' : 'wish_list_id submitted does not exist within the database.'
```

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 6]
```diff
@@ -1,11 +1,146 @@
...
+        return JsonResponse(response, safe=False)
+    else:
+        #Create new book entry in to the Wish_List_Books table on mysql. Then state its success.
+        addbook = WishListBooks.objects.create(
+            book_id = bookID,
+            wish_list_id = wishlistID
+        )
+
+        response = {
+            'Message' : 'Book ID has been successfully added to associated Wishlist.',
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:75]:
```suggestion
            'Message' : 'Book has been successfully added to the associated wish list.',
```

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 7]
```diff
@@ -1,11 +1,146 @@
...
+            'Book ID' : bookID,
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:76]:
```suggestion
            'book_id' : bookID,
```

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 8]
```diff
@@ -1,11 +1,146 @@
...
+            'Wish List ID' : wishlistID
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:77]:
```suggestion
            'wish_list_id' : wishlistID
```

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 9]
```diff
@@ -1,11 +1,146 @@
...
+        data = {}
+    
+    #Place the ID and wishlist into their associated variables.
+    userID = data.get('user_id')
+    wishlistName = data.get('wishlist_name')
+
+    #Check whether userID or wishlist name is missing. If so, send error. If not, continue to next check.
+    if userID == None or wishlistName == None:
+        response = {
+            'Error' : 'User ID or Wishlist Name is missing. Please enter it again.'
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:108]:
```suggestion
            'Error' : 'user_id or wish_list_name is missing. Please enter it again.'
```

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 10]
```diff
@@ -1,11 +1,146 @@
...
+
+    #Obtain user ID and their wishlist name.
+    try:
+        data = json.loads(request.body)
+    except json.JSONDecodeError:
+        data = {}
+    
+    #Place the ID and wishlist into their associated variables.
+    userID = data.get('user_id')
+    wishlistName = data.get('wishlist_name')
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:103]:
```suggestion
    wishlistName = data.get('wish_list_name')
```

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 11]
```diff
@@ -1,11 +1,146 @@
...
+    else:
+        #Create new wishlist with user ID and wishlist name.
+        WishLists.objects.create(
+            user_id = userID,
+            wishlist_name = wishlistName
+        )
+
+        response = {
+            'Message' : 'New wishlist created.',
+            'User ID' : userID,
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:142]:
```suggestion
            'user_id' : userID,
```

File: djangoBookStoreApi/views/wish_list_books_view.py [hunk 12]
```diff
@@ -1,11 +1,146 @@
...
+            'Wish List Name' : wishlistName
```
Comment (eandrews737) [inline djangoBookStoreApi/views/wish_list_books_view.py:143]:
```suggestion
            'wish_list_name' : wishlistName
```

**Review by eandrews737 — CHANGES_REQUESTED** [review]

**Review by lamad019 — COMMENTED** [review]
changes made

**Review by eandrews737 — DISMISSED** [review]

**Review by eandrews737 — APPROVED** [review]

**Review by junior033 — APPROVED** [review]

**Pull request merged**